### PR TITLE
Updates to searchable trait

### PR DESCRIPTION
**Pull request by willtj**
@JasparGupta commit messages should be descriptive enough to see what I'm doing here - main change is calling toSearchableArray on embedded models.

### CODE REVIEW

File: src/Searchable.php [hunk 1]
```diff
@@ -152,11 +164,11 @@ public function datesToSearchable(array $array): array
         foreach ($this->getArrayableRelations() as $key => $value) {
```
Comment (JasparGupta) [inline src/Searchable.php]: Just a small thing as I think it reads a bit better when used in `method_exists()`.

```suggestion
         foreach ($this->getArrayableRelations() as $key => $model) {
```

File: src/Searchable.php [hunk 2]
```diff
@@ -118,6 +114,22 @@ public function toSearchableArray()
     {
         $array = $this->toArray();
 
+        foreach ($this->getArrayableRelations() as $key => $value) {
```
Comment (JasparGupta) [inline src/Searchable.php]: I've done this backwards, sorry haha but here too.

```suggestion
        foreach ($this->getArrayableRelations() as $key => $model) {
```

**Comment by willtj** [comment]
I like the suggestion of having clearer variable names - I've gone with `$relation` since the value could be a model or a collection - LGTU?

**Comment by JasparGupta** [comment]
To be honest in my opinion, providing there is clear context I would use the name `$value`.
e.g.

`public function getOpeningDateAttribute($value){}`

Think its pretty obvious what `$value` is in this particular situation. (Plus, I also can't be asked to write `$openingDate` over and over again haha)

But yeah that makes perfect sense to me!

**Pull request merged**